pallets/elections/src/migrations/v0_to_v1.rs

```diff
@@ -73,3 +73,60 @@ pub fn migrate<T: Config, P: PalletInfoAccess>() -> Weight {
     StorageVersion::new(1).put::<P>();
     T::DbWeight::get().reads(reads) + T::DbWeight::get().writes(writes)
 }
+
+#[cfg(feature = "try-runtime")]
+pub fn pre_upgrade<T: Config, P: PalletInfoAccess>() -> Result<(), &'static str> {
+    match Members::<T>::get() {
+        Some(_) => {}
+        _ => return Err("No `Members` storage"),
+    }
+
+    if StorageVersion::get::<P>() == StorageVersion::new(0) {
+        Ok(())
+    } else {
+        Err("Bad storage version")
+    }
+}
+
+#[allow(dead_code)]
+#[cfg(feature = "try-runtime")]
+pub fn post_upgrade<T: Config, P: PalletInfoAccess>() -> Result<(), &'static str> {
```

Comment:
Suggestion: I would make these checks kinda stateful, i.e. I would return from `pre_upgrade` an object that should enforce validation in `post_upgrade`. In this case, I would pass old `Members` collection and check it against new `ReservedMembers`

Reply:
It should be possible to do (though we would need additional storage for this since pre and post-hooks do not operate on any mutable instance of some struct so they are stateless (if we don't look at the storage ofc). Anyway this can be done but in another task. This pr introduces the try-runtime feature and adds examples of the post/pre hooks for testing purposes of this feature. Refinement and improvement of these hooks should be done under different task